sbi/inference/posteriors/direct_posterior.py

```diff
@@ -384,7 +384,7 @@ def acceptance_at(x: Tensor) -> Tensor:
                         x, event_shape=self.posterior_estimator.condition_shape
                     )
                 },
-            )[1]
+            )[1].to(self._device)
```

Comment:
Why is this necessary? The proposal is `posterior_estimator,` which should be on the device and thus produce samples from it. I guess this is the "acceptance_rate", but this should be computed from the samples and thus also be on device.